### PR TITLE
Update RedHat 7 -> 8 -- release/6.0-staging

### PR DESCRIPTION
**Pull request by richlander**
Red Hat no longer supports .NET 6 on RHEL 7.

Context: https://access.redhat.com/support/policy/updates/net-core

Also resolving https://github.com/dotnet/runtime/pull/103380#discussion_r1680782274

@omajid @akoeplinger

### CODE REVIEW

**Review by omajid — APPROVED** [review]
I don't know the format of these files too well, but the general change from 7 to 8 makes sense to me.

**Review by akoeplinger — APPROVED** [review]

**Comment by carlossanlop** [comment]
No equivalent PR for `release/8.0-staging`?

File: eng/pipelines/coreclr/templates/helix-queues-setup.yml [hunk 1]
```diff
@@ -92,13 +92,13 @@ jobs:
         - (Debian.12.Amd64.Open)Ubuntu.2204.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-12-helix-amd64
         - (Debian.11.Amd64.Open)Ubuntu.2204.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-amd64
         - Ubuntu.2004.Amd64.Open
-        - RedHat.7.Amd64.Open
+        - RedHat.8.Amd64.Open
```
Comment (carlossanlop) [inline eng/pipelines/coreclr/templates/helix-queues-setup.yml:95]: @richlander this is failing:
```
.packages/microsoft.dotnet.helix.sdk/6.0.0-beta.24360.7/tools/Microsoft.DotNet.Helix.Sdk.MonoQueue.targets(59,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) Helix API does not contain an entry for RedHat.8.Amd64.Open
```

Here: https://dev.azure.com/dnceng-public/public/_build/results?buildId=745875&view=logs&j=1f44e919-17d8-5098-631c-31ef11f9e57b&t=d617d884-da52-5b82-bd49-f7613b408af4&l=92

and here: https://dev.azure.com/dnceng-public/public/_build/results?buildId=745875&view=logs&j=bc8f69e2-91b3-518c-84f8-466a04c1af40&t=0864dbdb-cb3f-5f82-b0b6-057b78db53e2&l=92

**Review by carlossanlop — CHANGES_REQUESTED** [review]
@richlander this is failing:
```
.packages/microsoft.dotnet.helix.sdk/6.0.0-beta.24360.7/tools/Microsoft.DotNet.Helix.Sdk.MonoQueue.targets(59,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) Helix API does not contain an entry for RedHat.8.Amd64.Open
```

Here: https://dev.azure.com/dnceng-public/public/_build/results?buildId=745875&view=logs&j=1f44e919-17d8-5098-631c-31ef11f9e57b&t=d617d884-da52-5b82-bd49-f7613b408af4&l=92

and here: https://dev.azure.com/dnceng-public/public/_build/results?buildId=745875&view=logs&j=bc8f69e2-91b3-518c-84f8-466a04c1af40&t=0864dbdb-cb3f-5f82-b0b6-057b78db53e2&l=92

**Comment by carlossanlop** [comment]
Approving for servicing as tell-mode. You can merge anytime if the CI looks good.

**Review by carlossanlop — APPROVED** [review]

**Comment by richlander** [comment]
Silly me. I didn't realize that we didn't have a RHEL 8 queue. Tracking issue: https://github.com/dotnet/dnceng/issues/3867.

I also didn't realize that they list was available: https://helix.dot.net/

**Comment by carlossanlop** [comment]
@richlander any updates on this? Is it ready to merge?

**Comment by richlander** [comment]
Thanks for checking in. The tracking issue is still open.

**Comment by carlossanlop** [comment]
BTW friendly reminder: Code Complete for the October Release is September 9. If we want this fix to be included in that release, please merge this PR _before_ that date. Otherwise, it will have to wait until November.

**Comment by carlossanlop** [comment]
Removing `servicing-approved` to prevent accidental merge. @richlander feel free to add the label again when you're ready to merge (it will unblock the check-service-labels required CI leg).

**Comment by carlossanlop** [comment]
@richlander we are going to need this PR merged soon as all existing PRs in the release/6.0-staging are now failing with the following error:

```
❌SENDHELIXJOB(0,0): error : Helix queue redhat.7.amd64.open is set for estimated removal date of 2024-10-01. 
In most cases the queue will be removed permanently due to end-of-life; please contact dnceng for any questions 
or concerns, and we can help you decide how to proceed and discuss other options.
```

**Comment by richlander** [comment]
I'll raise this with the right folks.

**Comment by richlander** [comment]
My thinking is:

- Replace RHEL7 with CentOS 7 to preserve validating min glibc.
- Add CentOS Stream 9 for outer loop testing.

**Comment by carlossanlop** [comment]
/ba-g The wasm "connection refused" and "Chrome was unable to start" failures that did not get linked to an issue are unrelated to this PR. One of them is https://github.com/dotnet/runtime/issues/104349

**Review by carlossanlop — APPROVED** [review]

**Pull request merged**